src/region/RegionList.tsx

```diff
@@ -4,26 +4,36 @@ import noop from 'lodash/noop';
 import RegionAnnotation from './RegionAnnotation';
 import useOutsideEvent from '../common/useOutsideEvent';
 import { AnnotationRegion } from '../@types';
-import { scaleShape } from './regionUtil';
 
 export type Props = {
     activeId?: string | null;
     annotations: AnnotationRegion[];
     className?: string;
     onSelect?: (annotationId: string | null) => void;
-    scale: number;
 };
 
-export function RegionList({ activeId, annotations, className, onSelect = noop, scale }: Props): JSX.Element {
+export function checkValue(value: number): boolean {
+    return value >= 0 && value <= 100; // Values cannot be negative or larger than 100%
+}
+
+export function filterRegion({ target }: AnnotationRegion): boolean {
+    const { shape } = target;
+    const { height, width, x, y } = shape;
+
+    return checkValue(height) && checkValue(width) && checkValue(x) && checkValue(y);
```

Comment:
Should we check for type as well?

Reply:
We actually do the `type` property check in RegionContainer, so the TS type is already refined when it reaches the actual components.